### PR TITLE
Added a new TranslationProvider for initializing translations for external use of NeetoUI

### LINKED ISSUE #2162: Support translations.
Found few instances for hardcoded string.

`BlockNavigation`
 - All of your unsaved changes will be lost. This can't be undone.
 - Discard changes.
 - You have unsaved changes
`ActionBlock`
 - Save changes
 - Cancel
`DatePicker`
 - Today
`Table`
 - Insert column right
 - Insert column left
 - Column info
 - Hide column
 - Delete column
`TimePicker`
 - Hour
 - Minute
 - Seconds
`TreeSelect`
 - No options
 
We should translate these when using other locale.

### PR DESCRIPTION
**Pull request by gaagul**
- Fixes #2162 

**Description**

- Added a new TranslationProvider for initializing translations for external use of NeetoUI

**Checklist**

- [x] I have made corresponding changes to the documentation.
- [x] I have updated the types definition of modified exports.
~~- [ ] I have verified the functionality in some of the neeto web-apps.~~
~~- [ ] I have added tests that prove my fix is effective or that my feature works.~~
~~- [ ] I have added proper `data-cy` and `data-testid` attributes.~~
- [x] I have added the necessary label (`patch`/`minor`/`major` - If package publish
      is required).

**Reviewers**

@gaagul _a

### CODE REVIEW

**Comment by gaagul** [comment]
@josephmathew900 _A Please review.

File: package.json [hunk 1]
```diff
@@ -16,7 +16,8 @@
   "files": [
     "dist",
     "types",
-    "*.d.ts"
+    "*.d.ts",
+    "src/translations"
```
Comment (josephmathew900) [inline package.json:20]: Add this in `exports` also:

```
    "./translations/*": {
      "import": "./src/translations/*.json",
      "require": "./src/translations/*.json"
    }
```

**Review by josephmathew900 — COMMENTED** [review]
@gaagul _a

**Comment by gaagul** [comment]
@josephmathew900 _A Please review the changes, I have also updated the storybook.

File: package.json [hunk 2]
```diff
@@ -199,6 +200,7 @@
     "react-colorful": "5.6.1",
     "react-dom": "18.2.0",
     "react-drag-listview": "2.0.0",
+    "react-i18next": "11.16.8",
```
Comment (josephmathew900) [inline package.json:203]: Add "react-i18next" to `peerDependencies` also

**Comment by josephmathew900** [comment]
@gaagul _a As we discussed in the call, please make the following changes:
- Users need to use TranslationProvider only if they wish to use a language other than English. This means we need to add a default value for all strings if i18n doesn't exist.
- If users need to add support for a new language, they need to create a PR and this needs to be mentioned in the documentation. We can remove the logic for supporting custom languages.
- And also create a translation index file that exports all the translations in neetoui. This way, if users use i18next they can merge these translations with their translations instead of using TranslationProvider.

**Comment by gaagul** [comment]
@josephmathew900 _a Please review the changes to the implementation.

File: stories/Introduction/TranslationProvider.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+export default App;
+
+```
+
+
+# Adding new language
+
+NeetoUI supports multiple languages internally, including English (en), Spanish (es), French (fr), German (de), Dutch (nl), Polish (pl), and Portuguese (pt).
+
+If your language is not in above list, feel free to create a locale package based on the [en](https://github.com/bigbinary/neeto-ui/blob/src/translations/en.json) language pack and send us a pull request.
```
Comment (josephmathew900) [inline stories/Introduction/TranslationProvider.stories.mdx:103]:
```suggestion
If your language is not in the supported list, feel free to create a locale package based on the [en](https://github.com/bigbinary/neeto-ui/blob/src/translations/en.json) language pack and send us a pull request.
```

File: stories/Introduction/TranslationProvider.stories.mdx [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+```
+
+The `TranslationProvider` component accepts the following props:
+
+- `language` (string): The language to use.
+  >> Note: The `language` prop is optional. If not provided, the default language is set to `en`.
+
+# Using translation files from NeetoUI
+
+NeetoUI supports multiple languages internally, including English (en), Spanish (es), French (fr), German (de), Dutch (nl), Polish (pl), and Portuguese (pt).
```
Comment (josephmathew900) [inline stories/Introduction/TranslationProvider.stories.mdx:55]:
```suggestion
```

File: stories/Introduction/TranslationProvider.stories.mdx [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+# Internationalization
+
+The default language of NeetoUI is currently English(en). If you wish to use other languages, follow the instructions below.
+
+## TranslationProvider
+
+The `TranslationProvider` component integrates internationalization (i18n) support for NeetoUI components in your React application using `react-i18next`.
+
+NeetoUI supports multiple languages internally, including English (en), Spanish (es), French (fr), German (de), Dutch (nl), Polish (pl), and Portuguese (pt).
```
Comment (josephmathew900) [inline stories/Introduction/TranslationProvider.stories.mdx:21]: Move this sentence under the Internationalization heading.

**Review by josephmathew900 — COMMENTED** [review]
@gaagul _a

**Comment by gaagul** [comment]
@josephmathew900 

Demo of exporting translations: https://gaagul-gigi.neetorecord.com/watch/571f9ac9-2398-4b30-a276-6f4a049fa14b

**Review by josephmathew900 — APPROVED** [review]
LGTM. @gaagul I have made some minor changes. Can you please verify the functionality and merge the PR?

**Comment by gaagul** [comment]
I have tested the changes, https://gaagul-gigi.neetorecord.com/watch/c9001105-3063-45dd-8bc4-13480f47b542.

**Pull request merged**